stablehlo/dialect/TypeInference.cpp

```diff
@@ -2544,7 +2544,15 @@ LogicalResult inferSelectOp(
 
 LogicalResult inferSelectAndScatterOp(
     Value operand, SmallVectorImpl<Type>& inferredReturnTypes) {
-  inferredReturnTypes.push_back(operand.getType());
+  auto operandRankedType = operand.getType().dyn_cast<RankedTensorType>();
+  if (!operandRankedType) {
+    inferredReturnTypes.push_back(operand.getType());
+  } else {
+    auto bounds = encodingToBounds(operandRankedType.getEncoding()).vec();
+    inferredReturnTypes.push_back(RankedTensorType::get(
+        operandRankedType.getShape(), operandRankedType.getElementType(),
+        boundsToEncoding(operandRankedType.getEncoding(), bounds)));
+  }
```

Comment:
Why do we need to change the original code at all? The `encodingToBounds` followed by `boundsToEncoding` seems to be a no-op?

Reply:
Good point, the bound information is carried by the type, so this separation of logic is not needed.